Plutarch/Repr/Data.hs

```diff
@@ -345,3 +325,57 @@ pmatchDataStruct (punsafeCoerce -> x) f = unTermCont $ do
       ds <- pletC $ psndBuiltin # x'
 
       pure $ groupHandlers (handlers ds) idx
+
+--------------------------------------------------------------------------------
+
+class (PLiftable a, PlutusRepr a ~ PLA.Data) => EachDataLiftable (a :: S -> Type)
+instance (PLiftable a, PlutusRepr a ~ PLA.Data) => EachDataLiftable (a :: S -> Type)
+
+class (a ~ AsHaskell b, PLiftable b, PlutusRepr b ~ PLA.Data) => ToAsHaskell (a :: Type) (b :: S -> Type)
+instance (a ~ AsHaskell b, PLiftable b, PlutusRepr b ~ PLA.Data) => ToAsHaskell (a :: Type) (b :: S -> Type)
```

Comment:
Do these need to be classes? They seem like collections of constraints, have no methods, and don't even seem tied to custom error messages.

Reply:
Yes, we can only partially apply type classes. Type synonyms can't be used on, let's say, `SOP.All`

Reply:
Ah, right, that annoying restriction. OK, that's fine then.